setup.cfg

```diff
@@ -3,6 +3,76 @@
 [bdist_rpm]
 doc-files = doc
 
+[options]
+python_requires = >=3.6
+install_requires =
```

Comment:
I noticed that there are a lot of duplicates between the three dependency groups (min, dev, and build-docs). Also, some seem to be in the wrong group(s). For example, `pytest`, `pytest-cov`, and `codecov` probably shouldn't be in the minimal dependency group.

Reply:
An initial proposal:
1. Remove `codecov`, `pytest`, `pytest-cov` from core dependencies.
2. Add `matplotlib`, `requests`, and maybe `plotly` to core dependencies.
3. Remove `coverage` and `flake8` from documentation dependencies.

Reply:
@NicolasGensollen @bthirion do you have opinions on this proposal? I could be missing the reasons behind the current division of dependencies.